deepcell/training.py

```diff
@@ -515,6 +513,61 @@ def loss_function(y_true, y_pred):
     print('batch size:', batch_size)
     print('validation_steps: ', total_test_pairs // batch_size)
 
+    # Make dicts to map the two generator outputs to the Dataset and model
+    # input here is model input and output is model output
+    features = sorted(features)
+
+    input_type_dict = {}
+    input_shape_dict = {}
+    for feature in features:
+
+        input_type_dict['{}_input1'.format(feature)] = tf.float32
+        input_type_dict['{}_input2'.format(feature)] = tf.float32
+
+        if feature == 'appearance':
+            app1 = tuple([None, train_data.min_track_length,
+                          train_data.crop_dim, train_data.crop_dim, 1])
+            app2 = tuple([None, 1, train_data.crop_dim, train_data.crop_dim, 1])
+
+            input_shape_dict['{}_input1'.format(feature)] = app1
+            input_shape_dict['{}_input2'.format(feature)] = app2
```

Comment:
The formatting of these names happens a lot in these blocks. Could we just define them once at the top and reuse it? I think it could really clean up some of this if/else block.

Reply:
Agreed. Moved as requested.